encoding/ssz/query/ssz_interface.go

```diff
@@ -0,0 +1,69 @@
+package query
+
+import (
+	"errors"
+)
+
+type SSZIface interface {
+	HashTreeRoot() ([32]byte, error)
+	MarshalSSZ() ([]byte, error)
+	UnmarshalSSZ([]byte) error
+	// HashTreeRootWith(hh *ssz.Hasher) error
+}
+
+// HashTreeRoot calls the HashTreeRoot method on the stored interface if it implements SSZIface.
+// Returns the 32-byte hash tree root or an error if the interface doesn't support hashing.
+func (info *sszInfo) HashTreeRoot() ([32]byte, error) {
+	if info == nil {
+		return [32]byte{}, errors.New("sszInfo is nil")
+	}
+
+	if info.iface == nil {
+		return [32]byte{}, errors.New("not stored iface")
+	}
+
+	// Check if the value implements the Hashable interface
+	if hashable, ok := info.iface.(interface{ HashTreeRoot() ([32]byte, error) }); ok {
+		return hashable.HashTreeRoot()
+	}
+
+	return [32]byte{}, errors.New("stored interface does not implement HashTreeRoot() method")
```

Comment:
```suggestion
	return info.iface.HashTreeRoot()
```
You can simply call `HashTreeRoot()` because you're guaranteed to have this method. That's the whole point of having the interface.

Reply:
Right, it was pointless to check the method once again. This is much cleaner!